diff --git a/lib/awestruct/handlers/css_tilt_handler.rb b/lib/awestruct/handlers/css_tilt_handler.rb
index b782cd3..4634514 100644
--- a/lib/awestruct/handlers/css_tilt_handler.rb
+++ b/lib/awestruct/handlers/css_tilt_handler.rb
@@ -4,7 +4,7 @@ require 'awestruct/handlers/front_matter_handler'
 require 'awestruct/handlers/layout_handler'
 
 require 'compass'
-require 'bootstrap-sass'
+#require 'bootstrap-sass'
 
 module Awestruct
   module Handlers
