diff --git a/spec/astruct_spec.rb b/spec/astruct_spec.rb
index 67d3e6a..3e06c42 100644
--- a/spec/astruct_spec.rb
+++ b/spec/astruct_spec.rb
@@ -1,5 +1,6 @@
 
 require 'awestruct/astruct'
+require 'awestruct/page'
 
 
 describe Awestruct::AStruct do 
diff --git a/spec/file_handler_spec.rb b/spec/file_handler_spec.rb
index b9919f1..6a65cc1 100644
--- a/spec/file_handler_spec.rb
+++ b/spec/file_handler_spec.rb
@@ -5,7 +5,9 @@ require 'awestruct/handlers/file_handler'
 describe Awestruct::Handlers::FileHandler do
 
   before :all do
-    @site = OpenCascade.new :encoding=>false, :dir=>Pathname.new( File.dirname( __FILE__ ) + '/test-data' )
+    @site = Hashery::OpenCascade.new
+    @site.encoding = false
+    @site.dir = Pathname.new( File.dirname( __FILE__ ) + '/test-data' )
   end
 
   it "should be able to read a valid absolute file handler" do
diff --git a/spec/front_matter_handler_spec.rb b/spec/front_matter_handler_spec.rb
index b72bdc5..2088ab6 100644
--- a/spec/front_matter_handler_spec.rb
+++ b/spec/front_matter_handler_spec.rb
@@ -6,7 +6,8 @@ require 'awestruct/handlers/front_matter_handler'
 describe Awestruct::Handlers::FrontMatterHandler do
 
   before :all do
-    @site = OpenCascade.new :encoding=>false
+    @site = Hashery::OpenCascade.new
+    @site.encoding = false
   end
 
   before :each do
diff --git a/spec/handler_chain_spec.rb b/spec/handler_chain_spec.rb
index 0e87690..7475ed7 100644
--- a/spec/handler_chain_spec.rb
+++ b/spec/handler_chain_spec.rb
@@ -18,7 +18,8 @@ describe Awestruct::HandlerChain do
 
 
   before :all do
-    @site = OpenCascade.new :encoding=>false
+    @site = Hashery::OpenCascade.new
+    @site.encoding = false
   end
 
   it "should use a regexp to match" do
diff --git a/spec/interpolation_handler_spec.rb b/spec/interpolation_handler_spec.rb
index 9be9f78..8a47d66 100644
--- a/spec/interpolation_handler_spec.rb
+++ b/spec/interpolation_handler_spec.rb
@@ -5,13 +5,15 @@ require 'awestruct/handlers/interpolation_handler'
 describe Awestruct::Handlers::InterpolationHandler do
 
   before :all do
-    @site = OpenCascade.new :encoding=>false
+    @site = Hashery::OpenCascade.new
+    @site.encoding = false
   end
 
   it "should interpolate content when rendered" do 
     handler = build_handler( 'This is #{cheese}' )
 
-    context = OpenCascade.new :cheese=>'swiss' 
+    context = Hashery::OpenCascade.new
+    context.cheese = 'swiss' 
     content = handler.rendered_content( context )
     content.should == 'This is swiss'
   end
@@ -20,7 +22,7 @@ describe Awestruct::Handlers::InterpolationHandler do
     if ::RbConfig::CONFIG['ruby_version'] =~ %r(^1\.9)
       input = 'url = url.replace(/\/?#$/, \'\');' 
       handler = build_handler( input )
-      content = handler.rendered_content( OpenCascade.new )
+      content = handler.rendered_content( Hashery::OpenCascade.new )
       content.should == input
     else
       pending "Cannot yet handle this test case with ruby 1.8"
diff --git a/spec/layout_handler_spec.rb b/spec/layout_handler_spec.rb
index 19de45c..6a2142e 100644
--- a/spec/layout_handler_spec.rb
+++ b/spec/layout_handler_spec.rb
@@ -14,7 +14,8 @@ describe Awestruct::Handlers::LayoutHandler do
 
 
   before :all do
-    @config = OpenCascade.new( :dir=>Pathname.new( File.dirname(__FILE__) + '/test-data/handlers' ) )
+    @config = Hashery::OpenCascade.new
+    @config.dir = Pathname.new( File.dirname(__FILE__) + '/test-data/handlers' )
     @engine = Awestruct::Engine.new
     @site = Awestruct::Site.new( @engine, @config )
     layout_loader = Awestruct::PageLoader.new( @site, :layouts )
diff --git a/spec/layouts_spec.rb b/spec/layouts_spec.rb
index 08a8f16..887ceca 100644
--- a/spec/layouts_spec.rb
+++ b/spec/layouts_spec.rb
@@ -5,12 +5,14 @@ require 'awestruct/handlers/file_handler'
 require 'awestruct/handlers/tilt_handler'
 
 require 'hashery/open_cascade'
+require 'set'
 
 describe Awestruct::Layouts do
 
   it "should be able to index layouts by simple name and output extension" do
     dir = Pathname.new( File.dirname( __FILE__ ) + '/test-data/handlers' )
-    site = OpenCascade.new( :dir=>dir )
+    site = Hashery::OpenCascade.new
+    site.dir = dir
     file_handler = Awestruct::Handlers::FileHandler.new( site, File.join( dir, 'haml-layout.html.haml' ) )
     haml_handler = Awestruct::Handlers::TiltHandler.new( site, file_handler )
     page = Awestruct::Page.new( nil, haml_handler )
\ No newline at end of file
diff --git a/spec/page_delegating_handler_spec.rb b/spec/page_delegating_handler_spec.rb
index 4aaaf85..f5a8815 100644
--- a/spec/page_delegating_handler_spec.rb
+++ b/spec/page_delegating_handler_spec.rb
@@ -7,7 +7,8 @@ require 'awestruct/handlers/page_delegating_handler'
 describe Awestruct::Handlers::PageDelegatingHandler do
 
   before :all do
-    @config = OpenCascade.new( :dir=>Pathname.new( File.dirname(__FILE__) + '/test-data/handlers' ) )
+    @config = Hashery::OpenCascade.new
+    @config.dir = Pathname.new( File.dirname(__FILE__) + '/test-data/handlers' )
     @engine = Awestruct::Engine.new( @config )
     @site = @engine.site
     layout_loader = Awestruct::PageLoader.new( @site, :layouts )
diff --git a/spec/page_spec.rb b/spec/page_spec.rb
index dc3e07f..240125e 100644
--- a/spec/page_spec.rb
+++ b/spec/page_spec.rb
@@ -14,8 +14,9 @@ describe Awestruct::Handlers::FileHandler do
   end
 
   before :all do
-    @site = OpenCascade.new :encoding=>false
-    @site.engine = OpenCascade.new
+    @site = Hashery::OpenCascade.new
+    @site.encoding = false
+    @site.engine = Hashery::OpenCascade.new
   end
 
   before :each do
diff --git a/spec/posts_archive_spec.rb b/spec/posts_archive_spec.rb
index 233275f..64860bc 100644
--- a/spec/posts_archive_spec.rb
+++ b/spec/posts_archive_spec.rb
@@ -53,7 +53,8 @@ describe Awestruct::Extensions::Posts do
 
   it "should assign default layout if specified to post without layout" do
     extension = Awestruct::Extensions::Posts.new( '/posts', :news, nil, nil, :default_layout => 'post' )
-    site = OpenCascade.new :encoding=>false
+    site = Hashery::OpenCascade.new
+    site.encoding = false
     page = __create_page( 2012, 8, 9, '/posts/mock-post.md' )
     page.stub(:layout).and_return(nil)
     page.should_receive(:layout=).with('post')
diff --git a/spec/support/shared_handler_example.rb b/spec/support/shared_handler_example.rb
index 498d66a..12d48ab 100644
--- a/spec/support/shared_handler_example.rb
+++ b/spec/support/shared_handler_example.rb
@@ -1,5 +1,6 @@
 require 'awestruct/engine'
 require 'awestruct/config'
+require 'awestruct/context'
 require 'rspec'
 
 require 'hashery/open_cascade'
@@ -14,7 +15,7 @@ shared_examples "a handler" do |theories|
   end
 
   def create_context
-    OpenCascade.new :site=>@site
+    Awestruct::Context.new(:site=>@site)
   end
 
   describe Awestruct::Handlers do
diff --git a/spec/yaml_handler_spec.rb b/spec/yaml_handler_spec.rb
index 7cc5686..c640daa 100644
--- a/spec/yaml_handler_spec.rb
+++ b/spec/yaml_handler_spec.rb
@@ -6,7 +6,8 @@ require 'awestruct/handlers/yaml_handler'
 describe Awestruct::Handlers::YamlHandler do
 
   before :all do
-    @site = OpenCascade.new :encoding=>false
+    @site = Hashery::OpenCascade.new
+    @site.encoding = false
   end
 
   it "should provide access to the yaml as front-matter" do 
